handbook/unity.adoc

```diff
@@ -0,0 +1,158 @@
+
+=== How to use Unity in UnetStack3 ?
+
+Unity is available as a premium agent and requires UnetStack 3.1 and higher. Configuring and using `Unity` agent to exploit spatial diversity is easy with just two simple steps:
+
+1. Setup the receiver nodes to cooperate.
+2. Add the Unity agent on the main receiver.
+
+Before we get into the details of how to set it up, let us start a simulated network with 3 nodes by simply running the `netq-network.groovy` simulation script. In the following description, we will setup the Node C to be the main receiver, Node B to be the assisting receiver and Node A to be the transmitter.
+
+==== Set up for receiver nodes to cooperate
+
+To set up the group of receivers to cooperate over a short-range network, we use Wormhole (<<Wormholes>>) agent provided in UnetStack. Transmitter node makes a transmission that is heard at all the receiver nodes. However, it is possible that none of the nodes are able to successfully recover the information received, as the communication link is noisy. In order to share the received noisy signals among the receivers, we can connect the receiver nodes using a Wormhole. A UDP connection between the two receiver nodes, over any IP based network (Ethernet, WiFi), can be established by adding just a few lines of code on the receiver nodes as shown below:
+
+[source]
+----
+container.add 'udp', new org.arl.unet.link.UdpLink()              //<1>
+container.add 'wormhole', new org.arl.unet.wormhole.Wormhole()    //<2>
+wormhole.dsp = 'udp'                                              //<3>
+----
+
+<1> Adds a UdpLink(<<Wired and over-the-air links>>) agent that implements a link protocol over UDP/IP for use over wired/wireless IP networks. 
+<2> Adds a Wormhole agent which allows the fjåge messages to be sent between containers over a Unet link. 
+<3> Tells the Wormhole agent to use the UdpLink as the Unet link to share information.
+
+Execute these three lines of code on both the assiting and main receiver (i.e., Node B and Node C).
+
+*Additional set up on the assisting receiver*:
+
+From the assisting receiver, we may not want to share the information from all agents, instead, we are interested in messages that are published only on the `Physical` agent’s topic. Therefore, we need an additional setting on the assiting receiver to forward only the messages received on the `Physical` agent’s topic to be sent over the Wormhole link.
+
+[source]
+----
+wormhole.publish = [topic(phy), topic(phy, Physical.SNOOP)]       //<1>
+----
+
+<1> Tells the wormhole agent to only forward the messages received on `Physical` agent's topic.
+
+Run the above one line of code on the assiting receiver (Node B).
+
+That’s all there is to set up the receiver nodes to share the information and cooperate (in this case over a `UdpLink`). With this set up any message received on the `Physical` agent’s topic on the assisting receivers will also be available at the main receiver.
```

Comment:
Add a note to indicate that the assisting receivers do not need the Unity agents? only the main receiver needs it?

Reply:
OK. I have now added a note for this.